zookeeper-server/src/test/java/org/apache/zookeeper/test/X509AuthTest.java

```diff
@@ -162,129 +149,136 @@ public String getFormat() {
-        public void checkValidity(Date date) throws CertificateExpiredException, CertificateNotYetValidException {
-        }
-        @Override
-        public int getVersion() {
-            return 0;
-        }
-        @Override
-        public BigInteger getSerialNumber() {
-            return null;
-        }
-        @Override
-        public Principal getIssuerDN() {
-            return null;
-        }
-        @Override
-        public Principal getSubjectDN() {
-            return null;
-        }
-        @Override
-        public Date getNotBefore() {
-            return null;
-        }
-        @Override
-        public Date getNotAfter() {
-            return null;
-        }
-        @Override
-        public byte[] getTBSCertificate() throws CertificateEncodingException {
-            return null;
-        }
-        @Override
-        public byte[] getSignature() {
-            return null;
-        }
-        @Override
-        public String getSigAlgName() {
-            return null;
-        }
-        @Override
-        public String getSigAlgOID() {
-            return null;
-        }
-        @Override
-        public byte[] getSigAlgParams() {
-            return null;
-        }
-        @Override
-        public boolean[] getIssuerUniqueID() {
-            return null;
-        }
-        @Override
-        public boolean[] getSubjectUniqueID() {
-            return null;
-        }
-        @Override
-        public boolean[] getKeyUsage() {
-            return null;
-        }
-        @Override
-        public int getBasicConstraints() {
-            return 0;
-        }
-        @Override
-        public byte[] getEncoded() throws CertificateEncodingException {
-            return encoded;
-        }
-        @Override
-        public void verify(PublicKey key) throws CertificateException, NoSuchAlgorithmException, InvalidKeyException, NoSuchProviderException, SignatureException {
-        }
-        @Override
-        public void verify(PublicKey key, String sigProvider) throws CertificateException, NoSuchAlgorithmException, InvalidKeyException, NoSuchProviderException, SignatureException {
-        }
-        @Override
-        public String toString() {
-            return null;
-        }
-        @Override
-        public PublicKey getPublicKey() {
-            return publicKey;
-        }
-        @Override
-        public X500Principal getSubjectX500Principal() {
-            return principal;
-        }
-        @Override
-        public Collection<List<?>> getSubjectAlternativeNames() {
-            List<Object> subjectAlternativeNamePair = new ArrayList<>();
-            subjectAlternativeNamePair.add(6);
-            subjectAlternativeNamePair.add(TEST_SAN_STR);
-            return Collections.singletonList(subjectAlternativeNamePair);
-        }
+  public static class TestCertificate extends X509Certificate {
+    @VisibleForTesting
+    static final String TEST_SAN_STR = "a:b:c(d;e;f)";
```

Comment:
Is indentation off here?

Reply:
Maybe some differences between auto formatters. I'll manually correct it.